plugins/obs-qsv11/common_utils_linux.cpp

```diff
@@ -55,33 +191,93 @@ mfxStatus simple_unlock(mfxHDL pthis, mfxMemId mid, mfxFrameData *ptr)
 mfxStatus simple_gethdl(mfxHDL pthis, mfxMemId mid, mfxHDL *handle)
 {
 	UNUSED_PARAMETER(pthis);
-	UNUSED_PARAMETER(mid);
-	UNUSED_PARAMETER(handle);
-	return MFX_ERR_UNSUPPORTED;
+	if (NULL == handle)
+		return MFX_ERR_INVALID_HANDLE;
+
+	// Seemingly undocumented, but Pair format defined by
+	// oneVPL-intel-gpu-intel-onevpl-23.1.0/_studio/mfx_lib/encode_hw/av1/linux/base/av1ehw_base_va_packer_lin.cpp
```

Comment:
I'm having trouble finding this repo. Could you link to it?

Reply:
I dont even think intel engineers could find it honestly. this is version specific and was pulled from the actual release tarball that my distro shipped.

https://spec.oneapi.io/onevpl/2.3.1/API_ref/VPL_structs_memory.html Here is the official docs on the type, notice its a "OS-Specific handle associated with a video frame", but actually what this means its its a https://github.com/intel/vpl-gpu-rt/blob/4170dd9fa1ea319dda81b6189616ecc9b178a321/_studio/mfx_lib/ext/mpeg2/include/mfx_mpeg2_encode_interface.h#L103 "surface pair" `mfxHDLPair`, at least in practice on linux.

And this encoded into the vaapi layer responsible for doing the accelerated frame handling https://github.com/intel/vpl-gpu-rt/blob/4170dd9fa1ea319dda81b6189616ecc9b178a321/_studio/shared/src/libmfx_core_vaapi.cpp#L1251 (srcHandle is a mfxHDLPair)

What actually goes in the pair you have to find in same file https://github.com/intel/vpl-gpu-rt/blob/4170dd9fa1ea319dda81b6189616ecc9b178a321/_studio/shared/src/libmfx_core_vaapi.cpp#L1464

intel's changing their code all the time so i could change it to one of these github links, assuming they dont rename their repositories. Cause im 70% sure this repo wasnt named vpl-gpu-rt when I was writing this code a few months ago...

https://spec.oneapi.io/onevpl/2.7.1/programming_guide/VPL_prg_mem.html Also this sample code seems oddly familiar to me, and wasnt there last I looked :)

Reply:
Added one of the github links too.